.github/workflows/slsa-goreleaser.yml

```diff
@@ -0,0 +1,163 @@
+        with:
+          name: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+          path: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+          if-no-files-found: error
+          retention-days: 5
+      - name: Release SBOM
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # v0.1.15
+        id: release-tag-with-sbom
+        with:
+          files: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+
+  # Generate and Upload SBOM Provenance to release
+  sbom-provenance:
+    needs: generate-sbom
+    permissions:
+      actions: read # Needed for detection of GitHub Actions environment.
+      id-token: write # Needed for provenance signing and ID
+      contents: write # Needed for release uploads
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.4.0
+    with:
+      base64-subjects: "${{ needs.generate-sbom.outputs.hash }}"
+      upload-assets: true
+
+  # Build
+  release:
+    permissions:
+      id-token: write # To sign the provenance.
+      contents: write # To upload assets to release.
+      actions: read # To read the workflow path.
+    strategy:
+      fail-fast: false
+      matrix:
+        goos: [linux, darwin, windows]
+        goarch: [amd64, arm64]
+        include:
+          - goos: linux
+            goarch: s390x
+          - goos: linux
+            goarch: ppc64le
+          - goos: linux
+            goarch: arm
+        exclude:
+          - goos: windows
+            goarch: arm64
+    needs: args
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.4.0
+    with:
+      config-file: .slsa-goreleaser-${{matrix.goos}}-${{matrix.goarch}}.yml
+      go-version: 1.19
+      evaluated-envs: "COMMIT_DATE:${{needs.args.outputs.commit-date}}, COMMIT:${{needs.args.outputs.commit}}, VERSION:${{needs.args.outputs.version}}, TREE_STATE:${{needs.args.outputs.tree-state}}"
+
+  # Sign
+  sign-build-artifact:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+      actions: read
+    strategy:
+      fail-fast: false
+      matrix:
+        goos: [linux, darwin, windows]
+        goarch: [amd64, arm64]
+        include:
+          - goos: linux
+            goarch: s390x
+          - goos: linux
+            goarch: ppc64le
+          - goos: linux
+            goarch: arm
+        exclude:
+          - goos: windows
+            goarch: arm64
+    needs: release
+    steps:
+      - name: Download the build artifact
+        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+        with:
+          name: helm-sigstore-${{matrix.goos}}-${{matrix.goarch}} # name of the binary defined at the .slsa-goreleaser-goos-goarch.yml files
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@9becc617647dfa20ae7b1151972e9b3a2c338a2b # v2.8.1
+        with:
+          cosign-release: 'v1.13.1'
```

Comment:
you can drop this then we always use the latest version available

Reply:
Sure, done

Reply:
I'm not sure that's true if the action is pinned by hash. `cosign-installer` sets a default value. You'd have to unpin the action in order to pick up new releases of cosign, but I'm not sure unpinning is a good idea.

https://github.com/sigstore/cosign-installer/blob/55fd288876ce156f3d39276ebf89f86bf3702f37/action.yml#L10-L13

Reply:
Interesting point, the cosign-installer should also be unpinned to do so, right? But then it would assume the risk of unpinned GHA dependency

Reply:
I think the real benefit likely comes from being able to utilize dependabot/renovate etc. to update the action (they won't update the `cosign-release` input). So not specifying the input would allow you to upgrade automatically via dependabot PRs.

So leaving it pinned with the input left off is probably best.